### PR TITLE
[DOC] Add watcher to the threadpool doc

### PR DESCRIPTION
**Pull request by renshuki**
`watcher` is a relatively common threadpool and I think it would be beneficial to add it to the doc.

Based on:
https://github.com/elastic/elasticsearch/blob/68817d7ca29c264b3ea3f766737d81e2ebb4028c/x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/Watcher.java#L536-L543

### CODE REVIEW

File: docs/reference/modules/threadpool.asciidoc
Comment (jrodewig) [inline docs/reference/modules/threadpool.asciidoc:111]:
```suggestion
`watcher`::
    For <<xpack-alerting,watch executions>>. Thread pool type is `fixed` with a
    size of `min(5 * (`<<node.processors, `# of allocated processors`>>`), 50)`.
    The queue_size is `1000`.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I'm going to incorporate a minor change. I'll wait for an engineer to approve before merging.

File: docs/reference/modules/threadpool.asciidoc
Comment (renshuki) [reply docs/reference/modules/threadpool.asciidoc]: Awesome, thank you @jrodewig 🙏

**Comment by joegallo** [comment]
I'll review this one today, thanks for the heads up @jrodewig.

**Review by joegallo — APPROVED** [review]
I added some commits by way of review, so I'm 👍 now. @jrodewig would you mind re-reviewing?

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks @renshuki @joegallo.

**Pull request merged**
